src/System.Collections.Concurrent/tests/Performance/ConcurrentQueuePerfTest.netcoreapp.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.Collections.Concurrent.Tests
+{
+    public class Perf_ConcurrentQueueTests
+    {
+        /// <summary>
+        /// Creates a list containing a number of elements equal to the specified size
+        /// </summary>
+        [Benchmark]
+        [InlineData(8, 1000)]
+        [InlineData(50, 1000)]
+        public void Enqueue_Dequeue(int threadsCount, int itemsPerThread)
+        {
+            var q = new ConcurrentQueue<int>();
+            Task.WaitAll((from i in Enumerable.Range(0, threadsCount) select Task.Run(() =>
```

Comment:
The number of tasks that actually run concurrently will be limited by thread pool heuristics, I suggest using dedicated threads instead or using long-running tasks that don't run on the thread pool:
```c#
Task.Factory.StartNew(() =>
{
   ...
}, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
```

Reply:
@kouvel Do you have an example (I don't write much C# core) ?  I copied this code from other multi-threaded corefx perf test.

Reply:
Could follow the general pattern in https://github.com/dotnet/corefx/blob/master/src/System.Collections/tests/Performance/Perf.List.cs. Each iteration could be everything inside the Task.WaitAll.

Reply:
I do not see any thread usage in the example you cite.  I'll use https://github.com/dotnet/coreclr/pull/13670 as a sample for thread usage.

Reply:
`SemaphoreSlimWaitDrainRate()` looks like a good sample

Reply:
Regarding Perf.List.cs, I just meant for that to be an example for a [Benchmark] test.

Many of the tests I wrote in https://github.com/dotnet/coreclr/pull/13670 have a high error margin and they're not all necessarily bad to regress if there's compensation elsewhere so I didn't make them official [Benchmark] tests that are verified by the CI. There's also a ConcurrentQueueThroughput test there.